core/templates/components/summary-tile/exploration-summary-tile.component.ts

```diff
@@ -29,52 +29,57 @@ import { DateTimeFormatService } from 'services/date-time-format.service';
-  @Input() getLastUpdatedMsec: number;
-  @Input() getNumViews: string;
-  @Input() getObjective: string;
-  @Input() getCategory: string;
-  @Input() getRatings: ExplorationRatings;
-  @Input() getContributorsSummary: string;
-  @Input() getThumbnailIconUrl: string;
-  @Input() getThumbnailBgColor: string;
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion, for more information see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  @Input() getCollectionId!: string;
+  @Input() getExplorationId!: string;
+  @Input() getExplorationTitle!: string;
+  @Input() getStoryNodeId!: string;
+  @Input() getLastUpdatedMsec!: number;
+  @Input() getNumViews!: string;
+  @Input() getObjective!: string;
+  @Input() getCategory!: string;
+  @Input() getRatings!: ExplorationRatings;
+  @Input() getContributorsSummary!: HumanReadableContributorsSummary;
+  @Input() getThumbnailIconUrl!: string;
+  @Input() getThumbnailBgColor!: string;
   // If this is not null, the new exploration opens in a new window when
   // the summary tile is clicked.
-  @Input() openInNewWindow: string;
-  @Input() isCommunityOwned: boolean;
-  // If this is true, collection preview tile for mobile
-  // will be displayed.
-  @Input() isCollectionPreviewTile: boolean;
+  @Input() openInNewWindow!: string;
+  @Input() getParentExplorationIds!: string;
   // If the screen width is below the threshold defined here, the mobile
   // version of the summary tile is displayed. This attribute is optional:
   // if it is not specified, it is treated as 0, which means that the
   // desktop version of the summary tile is always displayed.
-  @Input() mobileCutoffPx: number;
-  @Input() isPlaylistTile: boolean;
-  @Input() getParentExplorationIds: string;
-  @Input() showLearnerDashboardIconsIfPossible: string;
-  @Input() isContainerNarrow: boolean;
-  @Input() isOwnedByCurrentUser: boolean;
-
-  activityType: string;
-  resizeSubscription: Subscription;
-  explorationIsCurrentlyHoveredOver: boolean;
-  isWindowLarge: boolean;
-  userIsLoggedIn: boolean;
-  isRefresherExploration: boolean;
-  contributors: object;
-  lastUpdatedDateTime: string = '';
-  avgRating;
-  thumbnailIcon;
+  @Input() mobileCutoffPx!: number;
+  @Input() isCommunityOwned: boolean = false;
+  // If this is true, collection preview tile for mobile
+  // will be displayed.
+  @Input() isCollectionPreviewTile: boolean = false;
+  @Input() isPlaylistTile: boolean = false;
+  @Input() showLearnerDashboardIconsIfPossible!: string;
+  @Input() isContainerNarrow: boolean = false;
+  @Input() isOwnedByCurrentUser: boolean = false;
+
+  activityType!: string;
+  resizeSubscription!: Subscription;
+  explorationIsCurrentlyHoveredOver: boolean = false;
+  isWindowLarge: boolean = false;
+  userIsLoggedIn: boolean = false;
+  isRefresherExploration: boolean = false;
+  contributors!: object;
+  lastUpdatedDateTime: string | null = '';
```

Comment:
please add a comment explaining why `lastUpdatedDateTime` can be `null`

Reply:
done